### PR TITLE
Create CLI documentation with a Rust script (instead of Python) + style improvement

### PR DESCRIPTION
**Pull request by heeckhau**
I noticed that the CLI documentation was being generated using a Python script, so I rewrote it in Rust to make it easier to understand and more maintainable. Although it’s not a 1-to-1 conversion, I aimed to stay close to the original implementation to facilitate the review process. The output remains identical.

Additionally, I made a small change to the Markdown template (in a separate commit) to enhance the appearance of the generated documentation:
image

### CODE REVIEW

**Review by onbjerg — APPROVED** [review]
personally love this, we should give some love to the cargo script feature. curious what others think cc @mattsse @DaniPopes @rkrasiuk @Rjected

**Review by DaniPopes — APPROVED** [review]
Yeah, this is way better, thanks! Also needs to install nightly in CI

This was adapted from the Foundry book (<https://github.com/foundry-rs/book/blob/8f8215a10248f8e4d4870e3c03dd836d64169fd1/scripts/gen_output/help.py#L1>), would you mind also opening a PR there too?

File: book/cli/help.rs [hunk 1]
```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env cargo +nightly -Zscript
```
Comment (DaniPopes) [inline book/cli/help.rs:1]: Needs `-S` passed to `env` for multiple arguments

Comment (heeckhau) [reply book/cli/help.rs]: Added

**Review by Rjected — COMMENTED** [review]
epic, looks like the current book CI lint uses a stable toolchain, so you will have to install the nightly toolchain as well in the lint workflow

**Comment by heeckhau** [comment]
This change was merged in Foundry: https://github.com/foundry-rs/book/pull/1281
@DaniPopes I have updated the PR to include your changes.

I've also added Rust nightly to the GitHub workflow file

**Review by DaniPopes — APPROVED** [review]

**Review by Rjected — APPROVED** [review]

**Pull request merged**